In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [ ]:
SHOP = pd.read_csv('/content/drive/My Drive/Colab Notebooks/2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')
SHOP.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 

In [ ]:
SHOP.sort_values(by="created_at")

order_id  shop_id  user_id  order_amount  total_items payment_method  \
1862      1863       39      738           536            4           cash   
1741      1742       39      910           268            2           cash   
3228      3229       97      912           324            2           cash   
1267      1268       80      798           290            2    credit_card   
2689      2690       49      799           258            2    credit_card   
...        ...      ...      ...           ...          ...            ...   
2765      2766        9      708           236            2          debit   
4890      4891       63      853           136            1           cash   
244        245       32      928           202            2           cash   
211        212       22      712           292            2    credit_card   
145        146       72      906           640            4           cash   

              created_at  
1862  2017-03-01 0:08:09  
1741  2017-03-01 0:10:19  
3228  2017-03-01 0:14:12  
1267  2017-03-01 0:19:31  
2689  2017-03-01 0:22:25  
...                  ...  
2765  2017-03-30 9:22:41  
4890  2017-03-30 9:27:00  
244   2017-03-30 9:30:28  
211   2017-03-30 9:40:40  
145   2017-03-30 9:55:00  

[5000 rows x 7 columns]

In [ ]:
SHOP['created_at'] = pd.to_datetime(SHOP['created_at'])

In [ ]:
def get_day(input):
    return input.day
SHOP['Day'] = SHOP['created_at'].apply(get_day)

In [ ]:
D1 = SHOP.loc[SHOP['Day'] == 1]
D1

order_id  shop_id  user_id  order_amount  total_items payment_method  \
4            5       18      883           156            1    credit_card   
6            7       87      915           149            1           cash   
63          64       91      962           160            1          debit   
66          67       66      743           322            2           cash   
95          96       35      975           328            2    credit_card   
...        ...      ...      ...           ...          ...            ...   
4881      4882       25      884           130            1          debit   
4935      4936       11      789           368            2    credit_card   
4938      4939       77      847           312            2    credit_card   
4945      4946       35      765           164            1    credit_card   
4948      4949       12      845           402            2           cash   

              created_at  Day  
4    2017-03-01 04:35:11    1  
6    2017-03-01 21:37:57    1  
63   2017-03-01 19:05:37    1  
66   2017-03-01 10:40:40    1  
95   2017-03-01 06:27:42    1  
...                  ...  ...  
4881 2017-03-01 17:02:52    1  
4935 2017-03-01 08:08:06    1  
4938 2017-03-01 11:32:33    1  
4945 2017-03-01 11:17:04    1  
4948 2017-03-01 10:59:55    1  

[182 rows x 8 columns]

In [ ]:
D3 = SHOP.loc[SHOP['Day'] == 3]
D3

order_id  shop_id  user_id  order_amount  total_items payment_method  \
1            2       92      925            90            1           cash   
24          25       73      917           495            3           cash   
33          34        7      800           224            2    credit_card   
67          68       21      812           284            2          debit   
79          80       20      838           254            2    credit_card   
...        ...      ...      ...           ...          ...            ...   
4842      4843       70      761           519            3          debit   
4852      4853       69      957           393            3          debit   
4883      4884       27      896           507            3    credit_card   
4939      4940       36      719           130            1    credit_card   
4953      4954       13      956           320            2           cash   

              created_at  Day  
1    2017-03-03 17:38:52    3  
24   2017-03-03 13:01:03    3  
33   2017-03-03 05:31:15    3  
67   2017-03-03 15:52:53    3  
79   2017-03-03 14:00:25    3  
...                  ...  ...  
4842 2017-03-03 20:24:22    3  
4852 2017-03-03 22:45:43    3  
4883 2017-03-03 23:45:24    3  
4939 2017-03-03 17:18:54    3  
4953 2017-03-03 11:56:44    3  

[162 rows x 8 columns]

In [ ]:
D5 = SHOP.loc[SHOP['Day'] == 5]
D5

order_id  shop_id  user_id  order_amount  total_items payment_method  \
28          29       40      944           322            2           cash   
36          37       84      979           459            3    credit_card   
47          48       52      791           438            3           cash   
118        119       84      789           153            1    credit_card   
122        123       35      997           492            3          debit   
...        ...      ...      ...           ...          ...            ...   
4741      4742       47      725           290            2    credit_card   
4789      4790       23      744           312            2    credit_card   
4823      4824       44      702           288            2    credit_card   
4921      4922       44      858           432            3    credit_card   
4940      4941       43      992           181            1          debit   

              created_at  Day  
28   2017-03-05 02:12:53    5  
36   2017-03-05 22:44:34    5  
47   2017-03-05 15:50:51    5  
118  2017-03-05 04:08:32    5  
122  2017-03-05 01:16:57    5  
...                  ...  ...  
4741 2017-03-05 20:52:54    5  
4789 2017-03-05 11:29:07    5  
4823 2017-03-05 21:06:59    5  
4921 2017-03-05 17:21:41    5  
4940 2017-03-05 09:03:14    5  

[163 rows x 8 columns]

In [ ]:
print(D1['order_amount'].mean())
print(D3['order_amount'].mean())
print(D5['order_amount'].mean())

583.2637362637363
300.7962962962963
774.8036809815951


In [ ]:
SHOP['PPI'] = SHOP['order_amount'] / SHOP['total_items']
SHOP['PPI']

0       112.0
1        90.0
2       144.0
3       156.0
4       156.0
        ...  
4995    165.0
4996    117.0
4997    117.0
4998    177.0
4999    144.0
Name: PPI, Length: 5000, dtype: float64

In [ ]:
SHOP['PPI'].mean()

387.7428

When it was stated that the AOV was $3145.13, I thought it was a bit fishy. Whether or not tax was included in the AOV is not the important bit, but that number was made under the assumption that since each store only has one model of a particular shoe and out of a hundred, over the course of a month rougly brings you to 100 multipled by 30 stores AKA 3000 plus tax?

Well using the new feature of Day, made in part because I could not access it otherwise, I chopped it down to 182 rows figuring that that number must be the amount of transactions done by said stores and looked at the order amount. After comparing three days of a given week, while it can't say the same for the month, as one can see, they are radically different from each other. This would strike out the idea that each store would sell the same amount of a product.

**Think about what could be going wrong with our calculation. Think about a better way to evaluate this data.**

Well I think a good start is to get more general values that could be easily graphable should we able to separate the dataset into each individual shop. And while I could do that, it would take allot of room on this file which I really dont want to overcomplicate. Regardless, I made a PPI AKA Price Per Item, which took the amount made in a day and divided by how many items sold. I guess you could say this would be the AOV at least if we broke it down into specific store.

**What metric would you report for this dataset?**


**What is its value?**

I would still use AOV as the metric. However the value that I had gotten was, assuming we are using the sample dataset is $387.74.


Question 2: SQL Queries

How many orders were shipped by Speedy Express in total?

1. SELECT * FROM Orders gives us all the entries that were shipped by Speedy Express. This would give us 196 total orders.

What is the last name of the employee with the most orders?

2. 

SELECT EmployeeID, COUNT(EmployeeID) as TotalReps
FROM Orders
GROUP BY EmployeeID;

This outputs a table where it gives the total number of times each employee ordered something and that is counted in the TotalRep column.

The answer would be employee with ID # 4.

What product was ordered the most by customers in Germany?

SELECT p."ProductID", p."ProductName", SUM(o."quantity") AS quantity
FROM Customers, OrderDetails AS o
    INNER JOIN "Products" AS p
    ON o."ProductID" = p."ProductID"
WHERE Country = "Germany"
GROUP BY o."ProductID"
ORDER BY SUM(o."quantity") DESC, p."ProductName" ASC

In this case, the product that sold the most in Germany would be ProductID 31, Gorgonzola Telino at 5038 Units.
